### ISSUE TITLE
 Strange behavior using JUnit options when scenario failed.

### ISSUE THREAD

**Issue by L3K0V** [cucumber/cucumber-jvm#1015]
With `--allow-started-ignored` flag:
![ko](https://cloud.githubusercontent.com/assets/2267270/16118794/5c713438-33e1-11e6-97b9-dac13add82ac.jpg)

Without `--allow-started-ignored` flag:
![ok](https://cloud.githubusercontent.com/assets/2267270/16118878/b2812edc-33e1-11e6-9aca-1468b412fe83.jpg)

In first case times are okay, but the feature structure is broken, when disable it the result is reversed - no times but okay structure. Is there any chance to get both working - times and not broken feature hierarchy.

Example scenario:

```
@bosnia
  Scenario: Open fiscal receipt
    Given fiscal device in normal state
    But device is Bosnian
    And set IOSA wtih value "1122334455667788"
    When open fiscal receipt with operator 1 and password "0000"
    Then command must execute normally
```

Notice that in scenario I first check `But device is Bosnian` but JUnit report it after `And set IOSA wtih value "1122334455667788"`

Thanks.

**Comment by aslakhellesoy** [comment]
We'll aim to fix this in 2.0 - it's a long standing bug

**Comment by mpkorstanje** [comment]
With https://github.com/cucumber/cucumber-jvm/pull/1142 this has hopefully been fixed. Let is know if is broken.
